<?xml version="1.0" ?>

<launch>
  <env name="ROSCONSOLE_CONFIG_FILE" 
    value="$(find experiment_launcher)/log_config/custom_rosconsole.conf"/>
  
  <param name="/use_sim_time" value="true"/>

  <!-- Run the map server -->
  <arg name="map_file" default="$(find experiment_launcher)/map/map_small.yaml"/>
  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" />
  <!-- <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)">
    <param name="frame_id" value="/map" />
  </node> -->

  <!-- NESTED AMCL FOR ROBOT 1 -->
  <group ns="robot1">
    <arg name="tf_prefix" value="robot1_tf" />
    <arg name="nested_amcl/initial_pose_x" value="3.44" />
    <arg name="nested_amcl/initial_pose_y" value="-1.01" />
    <arg name="nested_amcl/initial_pose_a" value="3.11" />

    <!-- SINA -->
    <arg name="nested_amcl/initial_leader_pose_x" value="0.62" />
    <arg name="nested_amcl/initial_leader_pose_y" value="-0.7" />
    <arg name="nested_amcl/initial_leader_pose_a" value="1.58" />

    <include file="$(find experiment_launcher)/launch/includes/nested_amcl_one_robot.launch">
      <arg name="initial_pose_x" value="$(arg nested_amcl/initial_pose_x)"/>
      <arg name="initial_pose_y" value="$(arg nested_amcl/initial_pose_y)"/>
      <arg name="initial_pose_a" value="$(arg nested_amcl/initial_pose_a)"/>
      <!-- SINA -->
      <arg name="initial_leader_pose_x" value="$(arg nested_amcl/initial_leader_pose_x)"/>
      <arg name="initial_leader_pose_y" value="$(arg nested_amcl/initial_leader_pose_y)"/>
      <arg name="initial_leader_pose_a" value="$(arg nested_amcl/initial_leader_pose_a)"/>
      <arg name="odom_frame_id" value="/robot1_tf/odom"/>
      <arg name="base_frame_id" value="/robot1_tf/base_footprint"/>
      <arg name="global_frame_id" value="/map"/>
      <arg name="use_map_topic" value="true"/>
    </include>

    <include file="$(find experiment_launcher)/launch/includes/nested_move_base_one_robot.launch"/>
  </group>


  <group ns="robot2">
    <arg name="tf_prefix" value="robot2_tf" />
    <arg name="amcl/initial_pose_x" value="0.62" />
    <arg name="amcl/initial_pose_y" value="-0.7" />
    <arg name="amcl/initial_pose_a" value="1.58" />

    <include file="$(find experiment_launcher)/launch/includes/flat_amcl_one_robot.launch">
      <arg name="initial_pose_x" value="$(arg amcl/initial_pose_x)"/>
      <arg name="initial_pose_y" value="$(arg amcl/initial_pose_y)"/>
      <arg name="initial_pose_a" value="$(arg amcl/initial_pose_a)"/>
      <arg name="odom_frame_id" value="/robot2_tf/odom"/>
      <arg name="base_frame_id" value="/robot2_tf/base_footprint"/>
      <arg name="global_frame_id" value="/map"/>
      <arg name="use_map_topic" value="true"/>
    </include>

    <include file="$(find experiment_launcher)/launch/includes/flat_move_base_one_robot.launch"/>
  </group>

  <include file="$(find experiment_launcher)/launch/ground_truth_publisher.launch">
  </include>

</launch>
